src/annotation.ts

```diff
@@ -0,0 +1,159 @@
+        let response = await HttpClient.get_instance().get<AnnotationData>(
+            `/annotations/${annotation_pk}/`
+        );
+        return new Annotation(response.data);
+    }
+
+    static async create(handgrading_rubric_pk: number,
+                        data: NewAnnotationData): Promise<Annotation> {
+        let response = await HttpClient.get_instance().post<AnnotationData>(
+            `/handgrading_rubrics/${handgrading_rubric_pk}/annotations/`,
+            data
+        );
+        let result = new Annotation(response.data);
+        Annotation.notify_annotation_created(result);
+        return result;
+    }
+
+    static notify_annotation_created(annotation: Annotation) {
+        for (let subscriber of Annotation._subscribers) {
+            subscriber.update_annotation_created(annotation);
+        }
+    }
+
+    async save(): Promise<void> {
+        let response = await HttpClient.get_instance().patch<AnnotationData>(
+            `/annotations/${this.pk}/`,
+            filter_keys(this, Annotation.EDITABLE_FIELDS)
+        );
+
+        safe_assign(this, response.data);
+        Annotation.notify_annotation_changed(this);
+    }
+
+    async refresh(): Promise<void> {
+        let last_modified = this.last_modified;
+        let response = await HttpClient.get_instance().get<AnnotationData>(
+            `/annotations/${this.pk}/`
+        );
+
+        safe_assign(this, response.data);
+        if (last_modified !== this.last_modified) {
+            Annotation.notify_annotation_changed(this);
+        }
+    }
+
+    static notify_annotation_changed(annotation: Annotation) {
+        for (let subscriber of Annotation._subscribers) {
+            subscriber.update_annotation_changed(annotation);
+        }
+    }
+
+    async delete(): Promise<void> {
+        await HttpClient.get_instance().delete(
+            `/annotations/${this.pk}/`
+        );
+        Annotation.notify_annotation_deleted(this);
+    }
+
+    static notify_annotation_deleted(annotation: Annotation) {
+        for (let subscriber of Annotation._subscribers) {
+            subscriber.update_annotation_deleted(annotation);
+        }
+    }
+
+    static async get_order(handgrading_rubric_pk: number): Promise<number[]> {
+        let response = await HttpClient.get_instance().get<number[]>(
+            `/handgrading_rubrics/${handgrading_rubric_pk}/annotations/order/`
+        );
+        return response.data;
+    }
+
+    static async update_order(handgrading_rubric_pk: number, data: string[]): Promise<number[]> {
+        let response = await HttpClient.get_instance().put<number[]>(
+            `/handgrading_rubrics/${handgrading_rubric_pk}/annotations/order/`,
+            data
+        );
+        let result = response.data;
+        Annotation.notify_annotations_order_updated(result);
+        return result;
+    }
+
+    static notify_annotations_order_updated(annotation_list: number[]) {
```

Comment:
This notification function (and the corresponding update function) should also take in the pk of the handgrading rubric the annotations belong to.